ctest-next/templates/test.c

```diff
@@ -53,3 +53,27 @@ uint32_t ctest_signededness_of__{{ alias.id }}(void) {
     return all_ones < 0;
 }
 {%- endfor +%}
+
+{%- for item in ctx.field_size_offset_tests +%}
+
+// Return the offset of a struct/union field.
+uint64_t ctest_offset_of__{{ item.id }}__{{ item.field.ident() }}(void) {
+    return offsetof({{ item.c_ty }}, {{ item.c_field }});
+}
+
+// Return the size of a struct/union field.
+uint64_t ctest_size_of__{{ item.id }}__{{ item.field.ident() }}(void) {
+    return sizeof((({{ item.c_ty }}){}).{{ item.c_field }});
+}
+{%- endfor +%}
+
+{%- for item in ctx.field_ptr_tests +%}
+
+// Return a pointer to a struct/union field.
+// This field can have a normal data type, or it could be a function pointer or an array, which
+// have different syntax, so it's signature must be modified.
+typedef {{ item.volatile_keyword }}{{ item.c_signature }};
```

Comment:
Some update to the wording and `c_signature` is needed since this isn't a signature anymore